### PR TITLE
config: add SetDirectory

### PR DESCRIPTION
**Pull request by abursavich**
For the latest version of https://github.com/prometheus/prometheus/pull/7701

### CODE REVIEW

**Comment by abursavich** [comment]
Removed configtest package.

File: config/config.go [hunk 1]
```diff
@@ -32,3 +34,20 @@ func (s *Secret) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	type plain Secret
 	return unmarshal((*plain)(s))
 }
+
+// DirectorySetter is a config type that contains file paths that may
+// be relative to the file containing the config.
+type DirectorySetter interface {
+	// SetDirectory joins any relative file paths with dir.
+	// Any paths that are empry or absolute remain unchanged.
```
Comment (brian-brazil) [inline config/config.go:42]: empty

Comment (abursavich) [reply config/config.go]: done

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**